packages/back-end/src/jobs/cacheInvalidate.ts

```diff
@@ -0,0 +1,83 @@
+import Agenda, { Job } from "agenda";
+import { AWS_CLOUDFRONT_DISTRIBUTION_ID } from "../util/secrets";
+import AWS from "aws-sdk";
+import { CreateInvalidationRequest } from "aws-sdk/clients/cloudfront";
+
+const INVALIDATE_JOB_NAME = "fireInvalidate";
+type InvalidateJob = Job<{
+  urls: string[];
+  retryCount: number;
+}>;
+
+let agenda: Agenda;
+export default function (ag: Agenda) {
+  agenda = ag;
+
+  // Fire webhooks
+  agenda.define(INVALIDATE_JOB_NAME, async (job: InvalidateJob) => {
+    const { urls } = job.attrs.data;
+
+    if (!urls) return;
+
+    if (AWS_CLOUDFRONT_DISTRIBUTION_ID) {
+      const cloudfront = new AWS.CloudFront();
+      const params: CreateInvalidationRequest = {
+        DistributionId: AWS_CLOUDFRONT_DISTRIBUTION_ID,
+        InvalidationBatch: {
+          CallerReference: "" + Date.now(),
+          Paths: {
+            Quantity: urls.length,
+            Items: urls,
+          },
+        },
+      };
+
+      cloudfront.createInvalidation(params, function (err, data) {
+        if (err) {
+          console.log("Cache invalidate: Error invalidating CDN");
+          console.log(err, err.stack);
+          throw new Error("Cache invalidate: Error: " + err);
+        } else {
+          // successful response
+          console.log("Cache invalidate: suceeded");
+          console.log(data);
+        }
+      });
+    }
+    // add other invalidations here.
+  });
+  agenda.on(
+    "fail:" + INVALIDATE_JOB_NAME,
```

Comment:
I don't know if we care about retries for this.  I think the cache header is only 5 minutes right now, so it's not like if this fails the urls will stick around for a year.